src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/MethodObject.java

```diff
@@ -248,18 +256,23 @@ public MethodInvocationObject isDelegate() {
                                 }
                             }
                         }
-                    } else if (!PsiTreeUtil.findChildrenOfType(methodInvocationExpression, PsiReferenceExpression.class).isEmpty()) {
-                        Collection<PsiReferenceExpression> references = PsiTreeUtil.findChildrenOfType(methodInvocationExpression, PsiReferenceExpression.class);
-                        for (PsiReferenceExpression reference : references) {
-                            PsiElement resolvedReference = reference.resolve();
-                            if (resolvedReference instanceof PsiField) {
-                                PsiField psiField = (PsiField) reference.resolve();
-                                if (psiField != null && psiField.getContainingClass() != null && psiField.getContainingClass().equals(parentClass)
-                                        || psiField != null && parentClass != null && psiField.getContainingClass() != null && parentClass.isInheritor(psiField.getContainingClass(), true)) {
-                                    for (MethodInvocationObject methodInvocationObject : methodInvocations) {
-                                        if (methodInvocationExpression.equals(methodInvocation.getMethodExpression())) {
-                                            return methodInvocationObject;
-                                        }
+                    } else if (resolved instanceof PsiField) {
+                        PsiField variableBinding = (PsiField) resolved;
+
+                        if (variableBinding.getContainingClass().equals(parentClass) ||
```

Comment:
Please, keep in mind that `object.getSomething() `might be null.
If you are sure that  `object1` is not null and not sure about `object2`, then compare like this  `object1.equals(object2)`

Reply:
`parentClass` might be null, so add check for null

Reply:
done